compiler/luci/pass/src/PropagateConcatenationQparamPass.test.cpp

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "PropagateConcatenationQparamPassInternal.h"
+
+#include <luci/IR/CircleQuantParam.h>
+
+#include <vector>
+
+#include <gtest/gtest.h>
+
+namespace
+{
+
+void addQuantParam(luci::CircleNode &node, const std::vector<float> &scale,
+                   const std::vector<int64_t> &zp)
+{
+  assert(node.quantparam() == nullptr);
+
+  auto quantparam = std::make_unique<luci::CircleQuantParam>();
+  quantparam->scale = scale;
+  quantparam->zerop = zp;
+  node.quantparam(std::move(quantparam));
+}
+
+class SimpleConcatGraph
+{
+public:
+  SimpleConcatGraph()
+  {
+    concat_node.dtype(loco::DataType::U8);
+    concat_node.fusedActivationFunction(luci::FusedActFunc::NONE);
+    input_1.dtype(loco::DataType::U8);
+    input_2.dtype(loco::DataType::U8);
+
+    concat_node.values(0, &input_1);
+    concat_node.values(1, &input_2);
+
+    addQuantParam(concat_node, {3.14}, {77});
+    addQuantParam(input_1, {1.0}, {1});
+    addQuantParam(input_2, {2.0}, {2});
+  }
+
+  ~SimpleConcatGraph()
```

Comment:
Is it necessary to define the destructor and explicitly reset the pointers?

Reply:
Yes. If `input_1` or `input_2` is used by `concat_node` when the destructor is called, the below assertion occurs in `loco/src/IR/Node.cpp`.

```cpp
Node::~Node()
{
  // To detect dangling references
  assert(_uses.size() == 0);
}
```

Reply:
I see. This is probably because the nodes are not linked to Graph inputs / outputs.